chia/consensus/constants.py

```diff
@@ -26,37 +28,37 @@ class ConsensusConstants:
     SUB_EPOCH_BLOCKS: uint32  # The number of blocks per sub-epoch
     EPOCH_BLOCKS: uint32  # The number of blocks per sub-epoch, must be a multiple of SUB_EPOCH_BLOCKS
 
-    SIGNIFICANT_BITS: int  # The number of bits to look at in difficulty and min iters. The rest are zeroed
-    DISCRIMINANT_SIZE_BITS: int  # Max is 1024 (based on ClassGroupElement int size)
-    NUMBER_ZERO_BITS_PLOT_FILTER: int  # H(plot id + challenge hash + signage point) must start with these many zeroes
-    MIN_PLOT_SIZE: int
-    MAX_PLOT_SIZE: int
-    SUB_SLOT_TIME_TARGET: int  # The target number of seconds per sub-slot
-    NUM_SP_INTERVALS_EXTRA: int  # The difference between signage point and infusion point (plus required_iters)
-    MAX_FUTURE_TIME: int  # The next block can have a timestamp of at most these many seconds more
-    NUMBER_OF_TIMESTAMPS: int  # Than the average of the last NUMBER_OF_TIMESTAMPS blocks
+    SIGNIFICANT_BITS: uint8  # The number of bits to look at in difficulty and min iters. The rest are zeroed
```

Comment:
@arvidn Answering your comment https://github.com/Chia-Network/chia-blockchain/pull/11763#discussion_r888880357 here since its related to this PR:

`int` is not streamable no. Whats the risk? If you have concerns about the size of one let me know but i think they are all sufficiently scaled.

Reply:
the risk is that `int` has very different behavior compared to our fixed size integer types. for example `bytes(int)` vs. `bytes(uint64)` or `int.to_bytes()` vs. `uint64.to_bytes()`.

I would worry about some existing use of these constants silently changing behavior. It might require an audit of the places where we use them.

Reply:
It's a good point, perhaps there is a place where -1 is used somewhere, but I don't know of any from the top of my head. These are all positive constants.  Also, there are so many places where these are used, auditing would be very difficult.